test/components/plots/barPlotTests.ts

```diff
@@ -64,18 +64,18 @@ describe("Plots", () => {
         var bar1 = d3.select(bars[0][1]);
         assert.closeTo(TestMethods.numAttr(bar0, "width"), xScale.rangeBand(), 1, "bar0 width is correct");
         assert.closeTo(TestMethods.numAttr(bar1, "width"), xScale.rangeBand(), 1, "bar1 width is correct");
-        assert.equal(bar0.attr("height"), "100", "bar0 height is correct");
-        assert.equal(bar1.attr("height"), "150", "bar1 height is correct");
+        assert.strictEqual(+bar0.attr("height"), 100, "bar0 height is correct");
+        assert.strictEqual(+bar1.attr("height"), 150, "bar1 height is correct");
         assert.closeTo(TestMethods.numAttr(bar0, "x"), 111, 1, "bar0 x is correct");
```

Comment:
I am unsure if a precedent should be set here, but I can try to enforce a precedent after we do an audit over the testing framework.

After some discussion / thinking, I think we're running into a semantical difference in understanding this test.

bar0.attr("height") returns the "height" attribute of the bar0 selection.  However, this is not actually the height of the element.  It is literally just the value that is ripped from the DOM, which is actually returned as a string (since primitives are easily coercible to strings).

The actual height of the element comes after a layer of parsing the value that is gotten from the DOM.  The idea of a height is represented as a number, so we must parse this information and extract the number value from it.  This numerical value is actually the height of the bar.

Thus, this comes back to our understanding of the test message "bar1 height is correct".  If this means the height of the element, then we should be using the numerical value.  If this means the "height" attribute of the element, then we should be using the string value.

What do you think?